app/lib/features/spaces/widgets/space_list_widget.dart

```diff
@@ -1,65 +1,38 @@
 import 'package:acter/common/extensions/options.dart';
-import 'package:acter/common/toolkit/errors/error_page.dart';
 import 'package:acter/common/widgets/room/room_card.dart';
 import 'package:acter/features/space/widgets/space_sections/section_header.dart';
-import 'package:acter/features/spaces/providers/space_list_provider.dart';
-import 'package:acter/features/spaces/widgets/space_list_skeleton.dart';
 import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_gen/gen_l10n/l10n.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
-import 'package:logging/logging.dart';
-
-final _log = Logger('a3::space-list-widget');
 
 class SpaceListWidget extends ConsumerWidget {
-  final String? searchValue;
+  final ProviderBase<List<Space>> spaceListProvider;
   final int? limit;
   final bool showSectionHeader;
   final VoidCallback? onClickSectionHeader;
+  final String? sectionHeaderTitle;
+  final bool? isShowSeeAllButton;
+  final bool showSectionBg;
   final bool shrinkWrap;
   final Widget emptyState;
 
   const SpaceListWidget({
     super.key,
+    required this.spaceListProvider,
     this.limit,
-    this.searchValue,
     this.showSectionHeader = false,
     this.onClickSectionHeader,
+    this.sectionHeaderTitle,
+    this.isShowSeeAllButton,
+    this.showSectionBg = true,
     this.shrinkWrap = true,
     this.emptyState = const SizedBox.shrink(),
   });
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
-    final spaceLoader = ref.watch(spaceListSearchProvider(searchValue ?? ''));
-    return spaceLoader.when(
-      data: (spaceList) => buildSpaceSectionUI(context, spaceList),
-      error: (error, stack) => spaceListErrorWidget(context, ref, error, stack),
-      loading: () => const SpaceListSkeleton(),
-      skipLoadingOnReload: true,
-    );
-  }
-
-  Widget spaceListErrorWidget(
-    BuildContext context,
-    WidgetRef ref,
-    Object error,
-    StackTrace stack,
-  ) {
-    _log.severe('Failed to load spaces', error, stack);
-    return ErrorPage(
-      background: const SpaceListSkeleton(),
-      error: error,
-      stack: stack,
-      textBuilder: L10n.of(context).loadingFailed,
-      onRetryTap: () {
-        ref.invalidate(spaceListSearchProvider(searchValue ?? ''));
-      },
-    );
-  }
-
-  Widget buildSpaceSectionUI(BuildContext context, List<Space> spaceList) {
+    final spaceList = ref.watch(spaceListProvider);
```

Comment:
Remove bit of code by converting `spaceListProvider` from `FutureProvider` to simple `Provider`.